### PR TITLE
refine replication/parser and test case

### PR DESCRIPTION
**Pull request by holys**
1. use interface directly instead of pointer of interface(make it simple)
2. refine test case to make it robust

@siddontang  PTAL

### CODE REVIEW

File: replication/replication_test.go [hunk 1]
```diff
@@ -409,9 +412,14 @@ func (t *testSyncerSuite) TestMysqlBinlogCodec(c *C) {
 		return nil
 	}
 
-	err = p.ParseFile("./var/mysql.000001", 0, f)
+	dir, err := os.Open(binlogDir)
```
Comment (siddontang) [inline replication/replication_test.go]: need to close the dir?

Comment (holys) [reply replication/replication_test.go]: yes, a  mistake

Comment (holys) [reply replication/replication_test.go]: fixed

**Comment by holys** [comment]
PTAL again @siddontang

**Pull request merged**